uwblueprint/project-read-frontend
# Add Previous Enrolment Card Component to FamilyDetailsSidebar

### Notion ticket link



Previous enrolment section (https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=54f32176f7624da7acfec32dc7f9bfbf)



### Implementation description
- in the API types, added enrolments field (which gets the EnrolmentResponse) in the FamilyDetailResponse and students field in the EnrolmentResponse (so we can see which students are enrolled in various classes). 
- In the Family Sidebar, iterated through the enrolments for a given family and passed various attributes (session name, students, etc.) to the FamilyDetailEnrolmentCard component
- In FamilyDisebarEnrolmentCard, used useState to hide/show students on dropdown click



### Steps to test

1. Test with this backend: https://github.com/uwblueprint/project-read-backend/tree/prev-enrolment-section 
2. On either the registration or sessions page, click on a family row to open the Family Details Sidebar
3. Scroll to the bottom of the Family Details Sidebar
4. Verify that:
- the Previous Enrolments Section matches with the Figma Design
- Student names display when the dropdown is clicked (card expands)
- multiple enrolments show up if a family has been enrolled multiple times 



### What should reviewers focus on?
- Should Session (April 2021)/ Class(Friday & Sunday) be clickable?
![image](https://user-images.githubusercontent.com/43387820/128613527-2d614bac-3f73-4de1-a160-bb3c081ba8ca.png)
- Design: is the blue colour already implemented elsewhere in the app? (see comment on line 33 of FamilySidebarEnrolmentCard.tsx)
### Checklist
